tests/unit/s2n_rsa_pss_rsae_test.c

```diff
@@ -285,12 +294,23 @@ int main(int argc, char **argv)
                 &rsa_cert_chain->cert_chain->head->raw));
         EXPECT_EQUAL(rsa_pkey_type, S2N_PKEY_TYPE_RSA);
 
-        RSA *rsa_key = EVP_PKEY_get0_RSA(rsa_public_key.pkey);
+        /* Modify the rsa_public_key for the new test_case.
+         *
+         * In openssl 3 the function `EVP_PKEY_get1_RSA` returns a pre-cached copy of PKEY
+         * rather than the original copy. For this reason it is not possible to modify the
+         * original underlying RSA key.
+         *
+         * As a workaround, `rsa_key_copy` is modified, `rsa_pss_public_key` is set using
+         * `EVP_PKEY_set1_RSA` and finally rsa_key_copy is freed. This change is also
+         * backwards compatible with other ossl impl down to 1.0.2. */
```

Comment:
> I don't think going into the details of how the key gets initialized will help anyone trying to understand the test.
I am inclined to keep this comment in here, especially since this is test code and the extra context might help in the future. Specifically its a bit surprising to have to call both `RSA_set0_key` and `EVP_PKEY_set1_RSA` otherwise.

I could make it an `Aside` section to help with readibility.

RSA_set0_key(rsa_key_copy, n, e, d));
POSIX_GUARD_OSSL(EVP_PKEY_set1_RSA



```suggestion
        /* Modify the rsa_public_key for the new test_case.
         *
         * Aside:
         * In openssl 3 the function `EVP_PKEY_get1_RSA` returns a pre-cached copy of PKEY
         * rather than the original copy. For this reason it is not possible to modify the
         * original underlying RSA key.
         *
         * As a workaround, `rsa_key_copy` is modified, `rsa_pss_public_key` is set using
         * `EVP_PKEY_set1_RSA` and finally rsa_key_copy is freed. This change is also
         * backwards compatible with other ossl impl down to 1.0.2. */
```

Reply:
spoke offline: its prob best to remove the impl detail here since they are too specific. https://github.com/aws/s2n-tls/pull/3473/commits/635eaf499d6614e60d7d3cbed14ad023960eb4ae